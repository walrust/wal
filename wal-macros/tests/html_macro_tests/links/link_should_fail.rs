use wal_macros::html;

fn main() {
    html! { <Link /> };
    html! { <Link></Link> };
    html! { <Link to /> };
    html! { <Link to= /> };
    html! { <Link unsupported_attr="value" /> };
    html! { <Link to="value" unsupported_attr="value" /> };
    html! { <Link key="value" /> };
    html! { </Link> };
    html! { <Link> };
    html! { <Link to="value"> };
}
