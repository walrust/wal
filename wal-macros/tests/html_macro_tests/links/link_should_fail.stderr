error: `Link` requires `to` attribute
 --> tests/html_macro_tests/links/link_should_fail.rs:4:14
  |
4 |     html! { <Link /> };
  |              ^^^^

error: `Link` requires `to` attribute
 --> tests/html_macro_tests/links/link_should_fail.rs:5:14
  |
5 |     html! { <Link></Link> };
  |              ^^^^

error: expected `=`
 --> tests/html_macro_tests/links/link_should_fail.rs:6:22
  |
6 |     html! { <Link to /> };
  |                      ^

error: Expected a literal or an expression block
 --> tests/html_macro_tests/links/link_should_fail.rs:7:23
  |
7 |     html! { <Link to= /> };
  |                       ^

error: Unsupported attribute `unsupported_attr`. `Link` supports only `key` and `to` attributes
 --> tests/html_macro_tests/links/link_should_fail.rs:8:19
  |
8 |     html! { <Link unsupported_attr="value" /> };
  |                   ^^^^^^^^^^^^^^^^

error: Unsupported attribute `unsupported_attr`. `Link` supports only `key` and `to` attributes
 --> tests/html_macro_tests/links/link_should_fail.rs:9:30
  |
9 |     html! { <Link to="value" unsupported_attr="value" /> };
  |                              ^^^^^^^^^^^^^^^^

error: `Link` requires `to` attribute
  --> tests/html_macro_tests/links/link_should_fail.rs:10:14
   |
10 |     html! { <Link key="value" /> };
   |              ^^^^

error: This closing tag does not have a corresponding opening tag. (hint: try adding `<Link>`)
  --> tests/html_macro_tests/links/link_should_fail.rs:11:13
   |
11 |     html! { </Link> };
   |             ^^^^^^^

error: `Link` requires `to` attribute
  --> tests/html_macro_tests/links/link_should_fail.rs:12:14
   |
12 |     html! { <Link> };
   |              ^^^^

error: This start tag does not have a corresponding end tag. (hint: try adding `</Link>`)
  --> tests/html_macro_tests/links/link_should_fail.rs:13:13
   |
13 |     html! { <Link to="value"> };
   |             ^^^^^^^^^^^^^^^^^
